dev/token.js

```diff
@@ -0,0 +1,46 @@
+const readline = require('readline')
+const fs = require('fs')
+const { google } = require('googleapis')
+const credentials = require('./credentials.json')
+
+const SCOPES = [
+  'https://www.googleapis.com/auth/drive',
+  'https://www.googleapis.com/auth/documents'
+]
+const TOKEN_PATH = 'dev/token.json'
+
+function authorize(credentials) {
+  const { client_secret, client_id, redirect_uris } = credentials.installed
+  const oAuth2Client = new google.auth.OAuth2(client_id, client_secret, redirect_uris[0])
+
+  fs.readFile(TOKEN_PATH, (err, token) => {
+    if (err) return getNewToken(oAuth2Client)
+    oAuth2Client.setCredentials(JSON.parse(token))
+    console.log('\nYou can now use the playground: npm run playground')
+  })
+}
+
+function getNewToken(oAuth2Client) {
+  const authUrl = oAuth2Client.generateAuthUrl({
+    access_type: 'offline',
+    scope: SCOPES
+  })
+  console.log('Authorize this app by visiting this url:', authUrl)
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+  })
+  rl.question('Enter the code from that page here: ', code => {
+    rl.close()
+    oAuth2Client.getToken(code, (err, token) => {
+      if (err) return console.error('Error retrieving access token', err)
+      oAuth2Client.setCredentials(token)
```

Comment:
Could be managed with a async function awaited in authorize(). getNewToken should just get the new Token, and store it to the file.

Reply:
This code isn't optimized for now, more thoughts will be put into it if (or when) we move it into its own package